src/main/java/seedu/weme/logic/parser/MemeBookParser.java

```diff
@@ -6,26 +6,21 @@
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 
-import seedu.weme.logic.commands.AddCommand;
-import seedu.weme.logic.commands.ClearCommand;
 import seedu.weme.logic.commands.Command;
-import seedu.weme.logic.commands.DeleteCommand;
-import seedu.weme.logic.commands.EditCommand;
 import seedu.weme.logic.commands.ExitCommand;
-import seedu.weme.logic.commands.FindCommand;
 import seedu.weme.logic.commands.HelpCommand;
-import seedu.weme.logic.commands.ListCommand;
+import seedu.weme.logic.commands.MemesCommand;
 import seedu.weme.logic.parser.exceptions.ParseException;
 
 /**
- * Parses user input.
+ * Parses user input. Base context parser that all context parsers inherit from.
  */
-public class MemeBookParser {
+public abstract class MemeBookParser {
```

Comment:
This class doesn't look abstract. Is there a need to declare abstract at the moment?

Reply:
It’s just to prevent any MemeBookParsers from being created as it is. The only functioning parsers should be the context parsers